### PR TITLE
fix: org/re.po#123 links do not redirect to our togithub domain

### LINKED ISSUE #17284: Do we redirect the short form `org/repo#123` links to our `togithub` domain properly?
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

_No response_

### Please select which platform you are using if self-hosting.

_No response_

### If you're self-hosting Renovate, tell us what version of the platform you run.

_No response_

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I'm creating a new issue for a separate bug.

@HonkingGoose 

> By the way, we're not replacing the GitHub links with our `togithub` domain in the changelog section of the `prBody`.
> Check out the release notes for the update, they're full of links back to GitHub issues/PRs.
> - https://github.com/renovate-reproductions/17284/pull/3

@viceice:

> I think it's because they use the short form `org/repo#123`, maybe we don't detect them properly

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Yes, see: https://github.com/renovate-reproductions/17284

### PR DESCRIPTION
**Pull request by MaronHatoum**
## Changes
- fix regex to identify if repositories contains `.` in GitHub pull requests/discussions/issues URLs.
- reducing the URLs to \<org\>/\<repo\>#\<number\>

Check the release notes in this real repository example:
https://github.com/MaronHatoum/17284/pull/11

## Context
- Closes https://github.com/renovatebot/renovate/issues/17284

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — APPROVED** [review]

**Comment by rarkins** [comment]
I have a concern that this is making a large change to our link approach to solve only a small edge case. The impact of a mistake would be very high in terms of bad reputation, so we try to be very careful about changes to backlinks.

Before merging, I would like to ask this: if it causes any mass backlink spam by accident (e.g. a regression error in our backlink prevention) then do we have any way to detect this quickly without waiting for some outraged OSS maintainer to tell us?

**Comment by viceice** [comment]
@rarkins I don't see any a big risk to false backlinks generated by this pr

File: lib/modules/platform/github/massage-markdown-links.ts
Comment (PhilipAbed) [inline lib/modules/platform/github/massage-markdown-links.ts:14]: @viceice if we look at the changes here, this is the only change in the middle:
\/[a-zA-Z1-9\-_.]+\/
\/[-_a-z0-9]+\/ 
you approved this so hopefully you can explain to me
is it correct to remove capital letters backward slashes and dots(which not skipped for some reason)  here? is the URL always with small letters? or is it just so we won't catch any user names that have capital letters? 
the intention isn't clear to me here.

File: lib/modules/platform/github/massage-markdown-links.ts
Comment (PhilipAbed) [inline lib/modules/platform/github/massage-markdown-links.ts:17]: @viceice 
can't we just use the same URL above and add groups to it, for reducing?  why do we make here another regex

WDYT?

File: lib/modules/platform/github/massage-markdown-links.ts
Comment (PhilipAbed) [reply lib/modules/platform/github/massage-markdown-links.ts:14]: nevermind, there's insensitive flag at the end it means no need for big letters...

**Comment by PhilipAbed** [comment]
@rarkins do not merge this, im changing it.
adding an extra regex doesnt make sense at all

File: lib/modules/platform/github/massage-markdown-links.ts
Comment (PhilipAbed) [reply lib/modules/platform/github/massage-markdown-links.ts:17]: i dont see any reason to keep this regex

File: lib/modules/platform/github/massage-markdown-links.ts
Comment (viceice) [reply lib/modules/platform/github/massage-markdown-links.ts:14]: 😉  allowed usernames[^1]

[^1]: https://github.com/shinnn/github-username-regex

File: lib/modules/platform/github/massage-markdown-links.ts
Comment (viceice) [reply lib/modules/platform/github/massage-markdown-links.ts:17]: you can try to integrate it. 🤷‍♂️

Comment (PhilipAbed) [reply lib/modules/platform/github/massage-markdown-links.ts:17]: @viceice maroon made this regex because of
`(?<!api\.)`
this is failing validation on `regEx`, so he made a new almost identical regex without it

what is this? `(?<!api\.)` ???

Comment (PhilipAbed) [reply lib/modules/platform/github/massage-markdown-links.ts:17]: removing it

Comment (viceice) [reply lib/modules/platform/github/massage-markdown-links.ts:17]: it's look ahead regex

**Comment by PhilipAbed** [comment]
closing this as maroon is no longer working on it, opened https://github.com/renovatebot/renovate/pull/18944 instead

**Pull request closed without merging**
